### PR TITLE
Extend the existing example with alarms and recurrence-rule

### PR DESCRIPTION
**Pull request by bastianwegge**
As described in #612 I struggled a little and read parts of the awesome library code and the RFC to understand how to do alerting and recurrence. This PR aims to extend the existing example a little for upcoming visitors.

I saw that there is a `test_examples.py` that does what parts of the examples do but it looks a little different. If you want me to adjust the test in this PR, @jacadzaca  please feel free to drop a comment.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Minor changes:
 
 - Add funding information
 - Update windows to olson conversion for Greenland Standard Time
+- Extend examples with alarm + recurrence
```
Comment (stevepiercy) [inline CHANGES.rst]:
```suggestion
- Extend examples in Usage with alarm and recurrence
```

File: docs/usage.rst [hunk 2]
```diff
@@ -309,13 +309,33 @@ Add the event to the calendar::
 
   >>> cal.add_component(event)
 
+Extending the event with subcomponents you can create multiple alarms::
```
Comment (stevepiercy) [inline docs/usage.rst]:
```suggestion
By extending the event with subcomponents, you can create multiple alarms::
```

File: docs/usage.rst [hunk 3]
```diff
@@ -309,13 +309,33 @@ Add the event to the calendar::
...
+
+  >>> from icalendar import Alarm
+  >>> from datetime import timedelta
+  >>> alarm_1h_before = Alarm()
+  >>> alarm_1h_before.add("action", "DISPLAY")
+  >>> alarm_1h_before.add("trigger", timedelta(hours=-1))
+  >>> alarm_1h_before.add("description", 'Reminder: Event in 1 hour')
```
Comment (stevepiercy) [inline docs/usage.rst]:
```suggestion
  >>> alarm_1h_before.add("description", "Reminder: Event in 1 hour")
```

File: docs/usage.rst [hunk 4]
```diff
@@ -309,13 +309,33 @@ Add the event to the calendar::
...
+  >>> event.add_component(alarm_1h_before)
+
+  >>> alarm_24h_before = Alarm()
+  >>> alarm_24h_before.add("action", "DISPLAY")
+  >>> alarm_24h_before.add("trigger", timedelta(hours=-24))
+  >>> alarm_24h_before.add("description", 'Reminder: Event in 24 hours')
```
Comment (stevepiercy) [inline docs/usage.rst]:
```suggestion
  >>> alarm_24h_before.add("description", "Reminder: Event in 24 hours")
```

File: docs/usage.rst [hunk 5]
```diff
@@ -309,13 +309,33 @@ Add the event to the calendar::
...
+  >>> event.add_component(alarm_24h_before)
+
+Or even recurrence::
+
+  >>> event.add('rrule', {'freq': 'daily'})
```
Comment (stevepiercy) [inline docs/usage.rst:330]:
```suggestion
  >>> event.add("rrule", {"freq": "daily"})
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Just a few nitpicks, and LGTM. Thank you!

**Comment by bastianwegge** [comment]
@stevepiercy I just noticed that I used double-quotes in the new examples and the rest uses single quotes. I'm gonna quickly fix that as well.

**Comment by bastianwegge** [comment]
Feel free to merge or to request changes again. Thanks for your feedback!

**Review by stevepiercy — APPROVED** [review]
Consistency is good.

If this package adopts Black for formatting code and examples, though...

**Comment by stevepiercy** [comment]
I requested reviews from a few top maintainers who can do a release on PyPI. Thanks for the contribution!

**Review by jacadzaca — APPROVED** [review]

**Review by thet — APPROVED** [review]
This looks good to me. Thanks for the contribution!

**Comment by niccokunzmann** [comment]
Thanks!

**Pull request merged**